edge-client/edge_client.cpp

```diff
@@ -963,11 +963,11 @@ bool edgeclient_add_resource(const char *endpoint_name, const uint16_t object_id
     M2MResource *res;
 #ifdef MBED_EDGE_SUBDEVICE_FOTA
     if(object_id == SOFTWARE_COMPONENT) {
-        tr_info("creating fota software resources");
-        if (resource_name == NULL)
-            res = inst->create_dynamic_resource(String(res_name), "", resolved_resource_type, false, false, false);
-        else
+        tr_info("---creating fota software resources-----");
+        if(resource_name)
             res = inst->create_dynamic_resource(String(res_name), resource_name, resolved_resource_type, false, false, false);
+        else
+            res = inst->create_dynamic_resource(String(res_name), "", resolved_resource_type, false, false, false);
```

Comment:
@prakharbhatt007 lets check the resource_name before calling this function and if it is null then assign empty string. That way we dont have to do this if else twice in the code.